LamestWebserver/LamestWebserver/Core/MultiStream.cs

```diff
@@ -0,0 +1,196 @@
+                    if (Length > ret)
+                        ret = Length;
+                }
+                return ret;
+            }
+        }
+
+        /// <summary>
+        /// position of all streams
+        /// </summary>
+        public override long Position
+        {
+            get
+            {
+                long ret = 0;
+                foreach (Stream stream in Streams)
+                {
+                    if (Length > ret)
+                        ret = Length;
+                }
+                return ret;
+            }
+            set
+            {
+                foreach(Stream stream in Streams)
+                {
+                    stream.Position = value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// flush all streams
+        /// </summary>
+        public override void Flush()
+        {
+            foreach(Stream stream in Streams)
+            {
+                stream.Flush();
+            }
+        }
+
+        /// <summary>
+        /// read all streams
+        /// </summary>
+        /// <param name="buffer"></param>
+        /// <param name="offset"></param>
+        /// <param name="count"></param>
+        /// <returns></returns>
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            int bytesWritten = 0; 
+            foreach (Stream stream in Streams)
+            {
+                bytesWritten = stream.Read(buffer, offset, count);
+            }
+
+            return bytesWritten;
+        }
+
+        /// <summary>
+        /// Seek all streams
+        /// </summary>
+        /// <param name="offset"></param>
+        /// <param name="origin"></param>
+        /// <returns></returns>
+        public override long Seek(long offset, SeekOrigin origin)
+        {
+            long newPosition = 0;
+            foreach (Stream stream in Streams)
+            {
+               newPosition = stream.Seek(offset, origin);
+            }
+
+            return newPosition;
+        }
+
+        /// <summary>
+        /// Set Length of all streams
+        /// </summary>
+        /// <param name="value"></param>
+        public override void SetLength(long value)
+        {
+            foreach (Stream stream in Streams)
+            {
+                stream.SetLength(value);
+            }
+        }
+
+        /// <summary>
+        /// write all streams
+        /// </summary>
+        /// <param name="buffer"></param>
+        /// <param name="offset"></param>
+        /// <param name="count"></param>
+        public override void Write(byte[] buffer, int offset, int count)
+        {
+            foreach (Stream stream in Streams)
+            {
+                stream.Write(buffer, offset, count);
+            }
+        }
+
+        public override void Close()
+        {
+            foreach (Stream stream in Streams)
+            {
+                stream.Close();
+            }
+        }
```

Comment:
I'm a little worried about race-conditions in `Write` and `Close` - please make sure, that you cannot write to closed streams either from inside or outside this class.
Generally speaking: Usually I'd prefer internally, but could just assume, that everything would be perfectly threaded internally and make sure weird things don't happen (like with a `IsClosed` flag) externally.